File: DFS & BFS/boj2606.py
def dfs(graph, v, visited):
    visited[v] = True
    count = 1
    
    for neighbor in graph[v]:
        if not visited[neighbor]:
            count += dfs(graph, neighbor, visited)

    return count

n = int(input())
m = int(input())

graph = [[] for _ in range(n+1)]
visited = [False] * (n+1)

for _ in range(m):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)
    
print(dfs(graph, 1, visited)-1)

File: DFS & BFS/Boj2606.java
import java.util.ArrayList;
import java.util.Scanner;

public class Boj2606 {
    public static ArrayList<ArrayList<Integer>> network = new ArrayList<>();
    public static boolean[] visited;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int m = sc.nextInt();

        for(int i = 0; i <= n; i++) {
            network.add(new ArrayList<>());
        }
        visited = new boolean[n+1];

        for(int i = 0; i < m; i++) {
            int u = sc.nextInt();
            int v = sc.nextInt();
            network.get(u).add(v);
            network.get(v).add(u);
        }
        System.out.println(dfs(1) - 1);
    }

    public static int dfs(int x) {
        int count = 1;

        visited[x] = true;
        for(int neighbor : network.get(x)) {
            if(!visited[neighbor]) {
                count += dfs(neighbor);
            }
        }
        return count;
    }
}
